editorial review

dvdksn opened the pull request:
Here are some changes I would suggest making. In summary:

- Move the descriptions about common params, such as `mode`, from the registry page to the index page
- Rewrite the "Authentication" section to "Synopsis", and highlight the possible parameters that you can use. A bit like in the README.
- Bunch of minor changes...

Maybe it would be of interest to say something about the storage locations for each backend as well? We only mention it as an aside for `local`, but the Buildx README describes it in further detail.

jedevc commented on docs/guides/cache/azblob.md:
```diff
@@ -1,62 +1,46 @@
-> The `azblob` cache is currently unreleased. You can use it today, by using
-> the `moby/buildkit:master` image in your Buildx driver.
+> This cache backend is unreleased. You can use it today, by using the
+> `moby/buildkit:master` image in your Buildx driver. {: .warning }
```
Ah I did this so that we'd get the neat little view in GitHub for users who browse the docs through that:

![image](https://user-images.githubusercontent.com/7352848/191527782-5680a874-6daa-4097-9f29-bcb3ba756277.png)

Not quite sure if there's a way of getting it to show in GitHub and also docs.docker.com? @crazy-max :eyes:

crazy-max replied:
`> **Note**` works on docs.docker.com too so agree to keep it to have nice layout on GitHub and docs website.

crazy-max replied:
See 
* https://github.com/docker/buildx/blob/master/docs/guides/bake/compose-file.md#env-file
* https://docs.docker.com/build/customize/bake/compose-file/#env-file

dvdksn replied:
Ah, sorry, I didn't realize. Will add it back ofc!

jedevc replied:
I still see the ` {: .warning }` style in a few places, oops :eyes:

dvdksn replied:
oh I didn't realize we should take out the liquid tags. Will do.

jedevc commented on docs/guides/cache/azblob.md:
```diff
@@ -1,62 +1,46 @@
-$ docker buildx build --push -t <user>/<image> \
-  --cache-to type=azblob,account_url=https://myaccount.blob.core.windows.net,name=my_image \
-  --cache-from type=azblob,account_url=https://myaccount.blob.core.windows.net,name=my_image
+$ docker buildx build . --push -t <registry>/<image> \
```
This LGTM, I'm happy to have this be the standard way of writing this :tada:

We can make sure to copy it through to the rest of the pages as well later.

dvdksn replied:
ACK

jedevc commented on docs/guides/cache/index.md:
```diff
@@ -1,87 +1,179 @@
+Buildx supports the following cache storage backends:
 
-  ([guide](./inline.md))
+- [Inline cache](./inline.md) that embeds the build cache into the image.
```
I like this!

At some point I imagine us linking to a location in the reference docs when we have some more details to link to there, like in: https://github.com/docker/buildx/blob/master/docs/guides/drivers/index.md. Where would that link like that go in here, to keep consistency?

(note to self: if we change this here, we should change it everywhere in a follow-up)

dvdksn replied:
oh I see, yeah with multiple link targets it makes sense to stack them somehow. It just looked a bit odd/lonely with that one link to guides 😅 

like literally every technical writer ever, I have a thing for bullet lists, so maybe we could do something along these lines: (let's figure it out later when adding reference?)

<img width="1248" alt="image" src="https://user-images.githubusercontent.com/35727626/191586467-7d32c8cd-462a-4847-aa7d-ff524c581c37.png">

jedevc replied:
Yup, that sounds good, once we have exporter + cache backends + driver docs I think we can do a final "consistency pass".

jedevc commented on docs/guides/cache/index.md:
```diff
@@ -1,87 +1,179 @@
-  --cache-from type=registry,ref=<user>/<cache-image> .
+$ docker buildx build . --push -t <user>/<image> \
+  --cache-to type=<cache-type>[,parameters...] \
+  --cache-from type=<cache-type>[,parameters...]
```
I really like the idea of using this `[,parameters]` syntax :smile: I think we should probably use that across all the docs?

I think we should keep this as a concrete registry example if we can, I think it goes nicely with the multiple cache example just below.

jedevc commented on docs/guides/cache/index.md:
```diff
@@ -1,87 +1,179 @@
+While `min` cache is typically smaller (which speeds up import/export times, and
+reduces storage costs), `max` cache is more likely to get more cache hits.
+Depending on the complexity and location of your build, you should experiment
+with both parameters to get the results.
```
```suggestion
with both parameters to find the results that work best for you.
```

jedevc commented on docs/guides/cache/index.md:
```diff
@@ -1,87 +1,179 @@
+This property is only meaningful with the `--cache-to` flag. When fetching
+cache, BuildKit will auto-detect the correct media types to use.
+
+<!-- FIXME: link to image exporter guide when it's written -->
```
FIXME should be at the top of the configuration options section (since most of the parameters overlap).

jedevc reviewed (changes requested):
> Move the descriptions about common params, such as mode, from the registry page to the index page

This is much cleaner now, thanks :tada:

> Rewrite the "Authentication" section to "Synopsis", and highlight the possible parameters that you can use. A bit like in the README.

I think having a separate authentication section is still a good idea, some of the cache exporters use some non-trivial authentication mechanisms, and it would be good if we had the space to give some more examples, and elaborate a little more (as a follow-up to buildx we can add that content).

I like the synopsis heading though, splitting like that reads really well :tada:

> Bunch of minor changes...

All excellent :ok_hand: Some minor notes left.

dvdksn commented:
ok I added back the auth section, and I changed around the params section once more to make it more like the one in the buildkit readme -- I quite like the explicitness of showing what params apply to import and export cache respectively. PTAL!

jedevc commented on docs/guides/cache/azblob.md:
```diff
@@ -2,61 +2,57 @@
+- `account_url`: the base address of the blob storage account, for example:
+  `https://myaccount.blob.core.windows.net`
+- `name`: the name of the cache image.
+- `secret_access_key`: specifies the
```
*sorry*, super minor nitpick :smile: I think if the parameters are referenced in the authentication section later on, we should add a little note after them to point users in the right direction.

Also, I think it would look nice if we grouped shared parameters at the top of both lists? Not sure on that though.

I think `account_url` and `secret_access_key` are also `--cache-from` parameters as well here.

dvdksn replied:
hey, nice idea. I tried adding a common section for parameters. whaddya think?

jedevc replied:
I like the common parameters, it reads really well :+1:

jedevc commented on docs/guides/cache/gha.md:
```diff
@@ -2,108 +2,109 @@
+
+`--cache-from` options:
+
+- `scope`: cache scope (defaults to the name of the current Git branch).
```
`url` and `token` go here too.

jedevc commented on docs/guides/cache/s3.md:
```diff
@@ -1,64 +1,65 @@
-environment variables scheme for the [AWS Go SDK](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html).
+- `region`: geographic location
+- `bucket`: name of the S3 bucket used for caching
+- `name`: name of the cache image
```
Auth options go here too as well.
